proxy/src/schema.rs

```diff
@@ -44,24 +77,132 @@ impl Into<ProjectId> for IdInput {
     }
 }
 
-/// Encapsulates read paths in API.
-pub struct Query;
+/// Object representing a git branch.
+#[derive(Debug, Eq, Ord, PartialEq, PartialOrd, GraphQLObject)]
+struct Branch {
+    /// Name of the branch.
+    name: String,
+}
 
-#[juniper::object(Context = Context)]
-impl Query {
-    fn apiVersion() -> &str {
-        "1.0"
-    }
+/// Object representing a git tag.
+///
+/// We still need full tag support.
+#[derive(Debug, Eq, Ord, PartialEq, PartialOrd, GraphQLObject)]
+struct Tag {
+    /// Name of the tag.
+    name: String,
+}
 
-    fn projects(ctx: &Context) -> FieldResult<Vec<Project>> {
-        Ok(ctx.source.get_all_projects())
-    }
+/// Representation of a person (e.g. committer, author, signer) from a repository. Usually
+/// extracted from a signature.
+#[derive(GraphQLObject)]
+struct Person {
+    /// Name part of the commit signature commit.
+    name: String,
+    /// Email part of the commit signature commit.
+    email: String,
```

Comment:
Will we ever need to validate this? Maybe newtyping again?

Reply:
Which foreign type would you newtype here?